core/network/src/protocol/sync.rs

```diff
@@ -992,6 +1045,7 @@ impl<B: BlockT> ChainSync<B> {
 		}
 	}
 
+	/// Request header and justification for the given block number.
```

Comment:
There's nothing in this function to indicate this, but this is only used when we're doing ancestor search.

Reply:
There's a trace message :)

Reply:
Why is this only used in ancestor search? Is there no other case where you'd want to request header and justification?

Reply:
Ancestry request is always for a single block and is performed by hash. There's a number of similar functions (`download_new`, `download_stale`)  that configure `BlockRequest` in different ways for different purposes and also log them.